### DISCUSSION TITLE
Sprint 4

### DISCUSSION THREAD

**Discussion by Justus-at-Tazama** [tazama-lf/General-Issues#140] [General]
## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
None

## Minutes:

**@cshezi - Cebolenkosi Shezi**

**@vorsterk - Kyle Vorster**

**@Lenbkr - Len Bekker**

**@JeanPierreNell - Jean-Pierre Nell**

**@kawaki-san - Rodney Kanjala**

**@johanfol - Johan Foley**

**Comment by Justus-at-Tazama** [comment]
## 2023/05/22

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
None

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Worked on optimisation of Rule 001. Prepared the DetermineBandedOutcome function. Moved on to testing of the DBO function out of the library.

Today: Demo prep.

**@vorsterk - Kyle Vorster**

Thursday: Assisting JP with the deployment workflow for merging of code. Resolved a library conflict. Got the library to compile, but there was some kind of authentication issue. It wasn't working on the workflows we had in place. Pausing this until there's a ticket and moving back to the release. Turned on APM on SB4.

Today: Add APM on SB3. Carrying on with the release stuff.

**@Lenbkr - Len Bekker**

Yesterday: Comparison benchmarks to compare the Arango DB connection pooling. Bulk inserted 100k entries. Used randomised E2EId's for the lookups. Ran multiple iterations of various sizes. Running Arango without TLS for now. http could keep up with the Arango driver until it quit when the service started complaining due to max calls. Changing the max sockets improved speed. The sockets tested capped out at the number of threads that the server was able to handle.

Today: Reviewing sample rule for Foley. Demo prep.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: reviewed the deployment workfows with Kyle. Carried on with NATS. Completed outstanding merges for the conclusion of the sprint. Outstanding tickets closed. Fixed the typology processor bug.

Today: Carrying on with NATS. Sprint demo prep. Deploy typology processor to SB4.

**@kawaki-san - Rodney Kanjala**

Yesterday: Wrote documentation to justify passing filters to our generic functions. Optimise rule 020. Switched to rule 004.

Today: Demo prep. Rule 004, rule 020.

**@johanfol - Johan Foley**

Yesterday: Sample rule and template (rule executor) is done.

Today: Demo prep. Rule executor documentation. Look at standardising the DetermineOutcome function in the executor.

## Parking lot

NATS: Investigate using the library as a library. The library itself seems incapable of using NATS modules. There is a way to publish the NATS library to a local machine.

Typology Processor: Fixes the bug to use the network map for the routing. Justus to do E2E test.

Suggestions for the library: Previously we were making DB calls and for each call we were making some data validation. If we were making a check for the result, the error conditions were not specific to the query. Add standardised error conditions to the library for specific common conditions, e.g. data issues, validation issues, DB issues (timeouts, authentication, etc). YES! Foley will put errors into the scaffold where appropriate and the errors during the logic will be triggered from the rule itself, but also drawn from the library.

Testing: Updated Postman test generation scripts.

**Comment by Justus-at-Tazama** [comment]
## 2023/06/20

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
None

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Test suites for optimisation. How can we reuse the tests developed in rule 001?

Today: Carrying on with rule refactoring. Starting rule 003.

**@vorsterk - Kyle Vorster**

Yesterday: Automated deployment

Today: Automated deployment

**@Lenbkr - Len Bekker**

Yesterday: Wrapping up sprint tickets and demo prep. Moved some tickets to QA.

Today: Schedule a workshop for the Open Socket Zombie thing. Use the DA session. Finish outstanding stuff in Arango library and writing tests.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: NATS is communicating between services and sending traffic. Yesterday's issue was a folder/library misalignment issue. Employed a work-around, but needs a permanent fix.

Today: Investigate execution scope to address visibility of the right library - communication plan: Len, then Foley. Carrying on with NATS.

**@kawaki-san - Rodney Kanjala**

Yesterday: Rule optimisation. Rule 004 refactored to use a single query.

Today: Rule 004 testing, implementation of the datacache, 

**@johanfol - Johan Foley**

Yesterday: Rule runner.

Today: Lots of meetings.

## Parking lot

Testing via pipedream:

User story admin: Please admin your tickets quicker.

Rule runner: Figure out how to publish the private repository rules as a library for inclusion in the rule executor. Justus to create a ticket.

Arango Library: Should we deprecate unused queries? No new rule-specific queries are expected in the short term. Mark unused queries as deprecated. Make ticket for query append and batching to backlog for future work.

npm CI: can we add a package lock. Create a ticket. Foley says we can do this on the rule executor. Len needs this in the Husky pre-commit.

QA process: Please make sure our code is deployed to SB4 before dragging something into the QA column.

Sprint retro: Please fill in the sprint retrospective thing.

**Comment by Justus-at-Tazama** [comment]
## 2023/06/21

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
None

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Rule 003. Similar to rule 001.MIN and MAX function output in Arango is differently formatted.

Today: Rule 007, or the rule runner.

**@vorsterk - Kyle Vorster**

Yesterday: Got the Jenkins job to kick off the workflow. Excite! The workflow runs completely and creates the version. The job does not respond back to the workflow yet.

Today: Iterate over the release process and automation.

**@Lenbkr - Len Bekker**

Yesterday: Assisten JP with undefined problems in data preparation. Worked on outstanding functions in the library. Worked with Rodney to increase the library footprint. Created a ticket for future library work. Reviewed sprint log.

Len: Review the deployment for rule 003. Review rule 001 and the PR and take on a rule as well.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: NATS is working! Rolling it out to all the services.

Today: Fixing the library to make sure all the functions are using the library. Tweaking configurability around the .env file. Foley suggests that the .env should work the same way as for the rule processors. Add NATS to the remaining services. Review rule 901 to add NATS to the rule runner. (Review readme)

**@kawaki-san - Rodney Kanjala**

Yesterday: Worked on Rule 004. Generecised the rule config across all rule processors in the library. Rule parameters are not standard and uniform across rule configs. Created a branch and opened a PR - please review. Micro-optimisation in the use of the library. Started working on rule 006.

Today: Merge changes and start using the centralised rule config.

**@johanfol - Johan Foley**

Yesterday: meetings, rule 006, rule executor on SB4.

Today: Unit testing for the rule executor. Iteration over the rule runner.

## Parking lot

NATS, .env, library - Turn the issue into a ticket for the backlog.

Rule processors and NATS and rule runner. - JP will pack the NATS functionality into the rule runner when the end-to-end is complete

Implementing the rule runnner - for the rules already refactored, start wrapping them in the rule runner.

Rule optimisation - For the user stories, start at the first comment and not the original body.

**Comment by vorsterk** [comment]
2023/06/22
### **Attendees**
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

### **Apologies**
None

### **Minutes:**
@cshezi - Cebolenkosi Shezi

Yesterday: Convert rule 001 to work with rule executor , Picked up issues with band properties . Has different solutions which to choose.

Today: Rule executor for the error handling and data cache for rule 001 . Get answer for determine outcome function - Chat after standup with Foley.

@vorsterk - Kyle Vorster

Yesterday:  Helped JP with DP , Added more functionality on the release.yaml and was working on the tags and release notes. The release notes can also use the issues and milestones on the repo (Nice to have).

Today: Sort out the Release notes and the tags that for some reason aren't being pulled through. Then finish the added extras on Jenkins.

@Lenbkr - Len Bekker

Yesterday: Looking into the arango memory utilization specially the queues , arango has 3 different queues limit=4096 , Wrote a doc explain the results https://frmscoe.atlassian.net/wiki/spaces/FRMS/pages/54984720/Improve+Memory+Utilization . Rule 003 completed and deployed.

Today: Foley: one more check for wait for sync - Write 10 requests and run a lookup query to see if the 10 results come back . Delay write to disk properties.  Once done will pickup  a rule.

@JeanPierreNell - Jean-Pierre Nell

Yesterday: DP not population correctly Transaction history issue , waiting for Justus to test . Nats almost done with Library and testing  , closed other tickets with comments.

Today: Getting Lib done and Testing for Nats.

@kawaki-san - Rodney Kanjala

Yesterday: Rule config type discussion with Len and Justus , Input is always invalid to the determine outcome , Workflow files . Rule executor discussion with Shezi

Today: Rule type PR raised have a discussion with Foley after standup . Change more rules to lib and rule executor.

@johanfol - Johan Foley

Yesterday: Determine function for the band outcomes.

Today: Work with Shezi and Rodney for the determine outcome change.

### **Parking lot**

Shezi-  Bands to the Determine outcome function - Sitting after stand up

Rodney- Rule config exit conditions and parameters , parameters - Not set in stone more discussions on this because rule code will have to change. PR raised Foley will look at the open PR - Discussion after standup

**Comment by Justus-at-Tazama** [comment]
## 2023/06/23

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
None

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Completed the optimisation of rule 001. Deployment issues preventing "done". Breaking changes in the library. PR for review to Foley today. Publish the library for the rule and tested via the rule executor. Updates made to the executor.

Today: Finish up rule 001, continuing with rule 003. Foley suggests: handle the breaking change rule-side.

**@vorsterk - Kyle Vorster**

Yesterday: Got the tagging working with semantic versioning. Picked up a bug for combining patches and features in the same release. Fixed. Doing more tests. Redeployed Typology Processor and Data Preparation Service on SB4.

Today: Carry on with release packaging. Will work towards a demo on Monday.

**@Lenbkr - Len Bekker**

Yesterday: Looked into Arango optimisations. Reviewed documentation down to Rocks. Looked at the Arango benchmarking tool, but don't seem to meet our specific needs (Reads >> Writes). Maaaaybe there's a test we can use. Updated documentation for startup options. Increased the queue size 10x - resulting in connection issues. Increased buffer size - can't tell if it has an impact on memory during runtime, but does affect utilisation during startup. Arango benchmarking makes no sense. Larger queue and buffer size actually appears to slow queries down (slightly), but it appears to be more robust. Test-bed may be inadequate - CPU is the bottleneck and memory doesn't appear to feature.

Today: Review test scenarios for rule optimisation - see PR on rule 001.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Deployment issue on bug fixes resolved. Worked on the library to make it more secure. Compiling a new build, but turning up some Husky issues.

Today: Updating and publishing the library for NATS. Coming up CADProc, TADProc, CMS and Interdiction channels. Building the library is more effort than implementing it. Rule executor will be handled by Foley. Testing seems very straightforward - check if NATS has a best-practice guide for testing the services.

**@kawaki-san - Rodney Kanjala**

Yesterday: Rule 006 and 004. Completed refactoring on 004, pending tests. Rule 006 testing validation on query post-processing. Opened a draft PR on the rule executor in collab with Shezi. Rule config validation checks.

Today: Deploy rules 004 and 006. New rules.

**@johanfol - Johan Foley**

Yesterday: Spend some time with the team on handing over the rule executor. Updated the determineOutcome function. Upgraded rule 007.

Today: Adding tests, hopefully based on rule 001. Review PRs.

## Parking lot

determineOutcome: error was being thrown from the executor not from the code. It should not happen. Breaking change in the library and the rule executor did not know how to handled it.

Conflicting PRs on the rule executor: Use the one that Shezi did. Rodney will do a commit to Shezi's PR on some admin issues.

Rule testing: Basic ingestion and pass-through testing on a deployed processor must deliver a valid result.

Performance on Arango: Documentation, shortlist for additional tests. Write a query that simulates out benchmark pattern? Update documentation - postpone for now.

Rule optimsation - Hardcoded error messages will make regionalisation more difficult. Need an alternative solution.

**Comment by Justus-at-Tazama** [comment]
## 2023/06/26

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
None

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Submitted rule 001 for QA. Started on rule 003. Testing completed. PR created.

Today: Update error handling in rule 003, especially the config file. Config cannot be handled exclusively in the rule runner. Resubmit PR. Review testing results for rule 001.

**@vorsterk - Kyle Vorster**

Yesterday: Wrapping up the release stuff - some minor outstanding tasks to complete. Change log, logic to the release YAML. Testing at the moment. Not every tag was getting into the change log - now updated. We can use the release YAML files on the private repositories! (With additional release tokens.)

Today: Update documentation and preparing for a demo tomorrow. Need to speak to Foley about the release version we want.

**@Lenbkr - Len Bekker**

Yesterday: Arango stuff. Reviewed Rocks DB documentation to exhaustion and humbling. Documented findings so far. A missing suggestion from our implementation is to use ArangoDB's velocity packs. Improves payload size and performance. Reviewed rule 001 and rule 007. Some recommendations around best practice for npm packaging and security modifications.

Today: Starting rules development today. Rule 021.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Finished up fixing up linting issues on NATS. Fun! Looked at using a mock library for NATS Jetstream. Only found something old and undocumented. Rudimentary tests up and running to get the library published. Implemented a logger. Fixed some bugs on the previous implementation. Updates to rule processor to implement NATS. Fanning in - some edge cases around the order of stream connections and subscriptions.

Today: The method works great for pass-through. Fanning out from the CRSP is more complicated. There has to be a way for CRSP to have the multiple subjects from the network map. One other issue somewhere.

**@kawaki-san - Rodney Kanjala**

Yesterday: Trouble writing unit tests for refactored rules. Trying to find the right level of unit testing. Foley and Shezi's tests are pasta. Updated rules 004 and 006. Tests are compiling. Holding off on opening the PRs because the determineOutcome is duplicated from a testing perspective. Found a bug in the rule config. Fixed it. Rule config unwrap failed under certain conditions.

Today: Open PRs for 004 and 006. Review determineOutcome function and tests.

**@johanfol - Johan Foley**

Yesterday: Rule 007. Team support.

Today: determineOutcome function PoC for testing. Use throw/catch for rule processor .err handling. Throw the "reason". Exit conditions are still handled in the rule as valid rule outcomes that continue to the typology processor.

## Parking lot

Rule processor cache handling: Fix in the library to prevent an empty result from the DB to be cached in Redis. May also affect data preparation.

Rule processor testing:

**Comment by Justus-at-Tazama** [comment]
## 2023/06/27

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
None

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Rule 001 updated and back in QA. determineOutcome error condition escalated to the rule executor. PR loaded and reviewed. Merged and published. Issue in ensuring that the RE uses the most recent version of the rule processor. Updated the rule processor version in Jenkins to pull through. Completed rule 003. Completed rule 008.

Today: Merge/Publish rule 003. Create PR for rule 008. Starting rule 020 then 25.

**@vorsterk - Kyle Vorster**

Yesterday: Spoke to Foley about versioning. Feature version - 1.2.0. No need to include the private repositories into the release. Helm version changes. Git Pages to be automatically generated. Updated documentation. Testing and fixing.

Today: Complete YML files. Complete documentation. Complete Jenkins pipeline. Demo tomorrow!

**@Lenbkr - Len Bekker**

Yesterday: Worked on rule 021. Clarity required on spec. Tolerance is a percentage < 1. Assisted on CRSP issues in NATS.

Today: Updating tests. Complete rule 021, next rule.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: NATS. Challenges in updating the CRSP - mostly resolved. Need to use the library to be efficiently implemented. Merged Foley's request. The Merge request removes Arango and Redis. Foley says Redis were left in there for a reason. JP says the code in CRSP is so archaic it is unable to reference the library.

Today: Wrap up CRSP. Deploy locally to test end-to-end flow. (Rule executor for Foley).

**@kawaki-san - Rodney Kanjala**

Yesterday: OPened two PRs for Shezi to review. Additional test coverage required. Updated tests. Version conflicting in rules question. Started rule 016. Started rule 018. Spec issue - Justus to check.

Today: Complete rule 016 with the rule executor. Complete rule 018.

**@johanfol - Johan Foley**

Yesterday: Update RE and the CRSP for JP. Bug fix on the CRSP.

Today: Boost NATS. Implement on Rooley.

## Parking lot

CRSP/Redis/Arango/Library: Deferred to 1-2-1

Rule processor version conflicts: When a new release is published, Jenkins picks up the package. A release version in the private rules does not make sense at the moment. Park for future discussion; packages are good enough for the private repos for now. Status quo for now.

Pick up paired rules at the same time if able.

**Comment by Justus-at-Tazama** [comment]
## 2023/06/28

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
None

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Completed rule 008. PR up for review. Rule 003 updated. 

Today: Rule 025 and 027. Complete rule 003 version rollback.

**@vorsterk - Kyle Vorster**

Yesterday: Release automation still in progress. Added all release repositories. Working on getting Helm pages deployed. Everything else appears to be working!

Today: Carrying on, with carrying on.

**@Lenbkr - Len Bekker**

Yesterday: Rule 021. Complete development. Performed additional cleanup on the rule processor repository. Eliminated several dependencies.

Today: Push and package rule 021. Picked up the Redis TTL bug.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Fixed issues from yesterday! We have a published version we can implement. Working on getting the end-to-end working.

Today: Testing, bug-fixing, classing.

**@kawaki-san - Rodney Kanjala**

Yesterday: Working on rule 006. Error in the spec, and bug. Fixing, redeploying. PR awaiting review. Helped JP with git issues. Reviewed Shezi's PRs.

Today: Redeploy rule 006. Wrapping up other rules.

**@johanfol - Johan Foley**

Yesterday: Implemented NATS on Rooley. Rule 007 fixes.

Today: Figure out a way to test.

## Parking lot

CRSP design challenge:

**Comment by Justus-at-Tazama** [comment]
## 2023/06/29

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
None

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Works on local, but not deployed. Authorization failure. Please help. Rule 008 and rule 025 both have the issue.

Today: Finish up the rules with support, then new rule.

**@vorsterk - Kyle Vorster**

Yesterday: Helm is working. All the test cases for the release are working. Put together some screenshots. Added milestones.

Today: Finalising testing. Creating documentation for release test cases.

**@Lenbkr - Len Bekker**

Yesterday: Rule 021 complete and tested. Worked on the npm.ignore issue with Rodney. @Kyle: we're modifying the pipeline. Reviewed the rule config Redis caching bug. Added a fix to check the contents of the Arango response before updating the cache. Awaiting review and deployment. Streamlined the use of the Arango library in the rule processor.

Today: New rule today! (After deploying bug fix. Help Shezi.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: Worked on NATS pub/sub on many-to-many flows. Each rule processor will connect to the same stream and add their subjects to the same stream. Wrapped up CRSP and Typology Processor. Finishing CADProc and TADProc.

Today: Integrate rule processor. Deploy on Sandbox 1. Look for Helm charts to publish NATS. Get the end-to-end running.

**@kawaki-san - Rodney Kanjala**

Yesterday: Len and Rodney debugged an npm.ignore file. Wiorked with Shezi to resolve linting issues with Shezi. Refactored rule 004 - Len rejected one of the test cases in review. Rule 004 deployed, awaiting testing. Started with rule 16.

Today: deploy rule 016, start the next one up.

**@johanfol - Johan Foley**

Yesterday: NATS and project reporting. Testing on the rule executer.

Today: Review release with Kyle. Work on NATS with JP. Complete rule 007.

![image](https://github.com/frmscoe/General-Issues/assets/123470803/1f781233-56b1-4188-96a1-cd58cfbecdc6)

## Parking lot

NATS: deployment of rule processors should be transparent to the CRSP and the Typology Processor. NATS is in the release next week.

Helm charts don't have the configuration of Jenkins in it. Version Jenkins repository for the jobs. Create variables for each of the rule processors. Design a process that will allow the deployment of the private rule processor libraries with the platform itself. - Use @latest for the rule executer.

**Comment by Justus-at-Tazama** [comment]
## 2023/06/30

## Attendees
@johanfol - Johan Foley
@Lenbkr - Len Bekker
@JeanPierreNell - Jean-Pierre Nell
@cshezi - Cebolenkosi Shezi
@vorsterk - Kyle Vorster
@kawaki-san - Rodney Kanjala

## Apologies
None

## Minutes:

**@cshezi - Cebolenkosi Shezi**

Yesterday: Completed two rules. 025 and 027 in QA. Started on rule 030.

Today: Fixes and PRs for 025/027. 030 to QA, then new rules.

**@vorsterk - Kyle Vorster**

Yesterday: Fixing a few bugs on the release for the Helm stuff. We're pulling in the latest release tag from each typology as foretold by the prophecy.

Today: Complete deployment process today. Ready for the demo on Monday!

**@Lenbkr - Len Bekker**

Yesterday: Worked on two rules. Fixed a bug in the library. One of the indexes weren't hitting for rule 020. Rule 020 and 048 almost done.

Today: Publish two rules. Need input from Justus.

**@JeanPierreNell - Jean-Pierre Nell**

Yesterday: When the typology processor subs to NATS, it needs to listen to multiple inputs from rules. Functionality required has not been published  yet (two month horizon). There is an Alpha branch, but we parked our code in the meantime until the feature is launched. Finished up the CRSP and TADProc and TADProc. Pulling in a new high-volume merge for the rule processor.

Today: Complete the end-to-end!

**@kawaki-san - Rodney Kanjala**

Yesterday: 3 rules deployed and tested (004, 006, 016). Worked on enforcing a schema for dates in the database to make sure dates are stored correctly in Arango. Completed rule 018 - PR is up for review.

Today: Pick up some more rules today. 75 and 78 on deck.

**@johanfol - Johan Foley**

Yesterday: Rooley enhancements. CADProc improvements. Picked up a bug for the rule processor response. 007 done and deployed!

Today: Rule executor enhancements and NATS!

![image](https://github.com/frmscoe/General-Issues/assets/123470803/de312873-cff6-4790-b841-1a8d284e70da)

## Parking lot
